Source/WebCore/html/HTMLTextFormControlElement.cpp

```diff
@@ -907,6 +907,13 @@ void HTMLTextFormControlElement::adjustInnerTextStyle(const RenderStyle& parentS
         }
     }
 
+    if (parentStyle.fieldSizing() == FieldSizing::Content) {
+        if (textBlockStyle.isHorizontalWritingMode())
+            textBlockStyle.setMinWidth(Length { 1, LengthType::Fixed });
```

Comment:
I think this needs to use `caretWidth()` rather than 1, since the width can be 2 on Cocoa platforms.

Reply:
Does this override whatever min-width the web developer has set? If the developer set `min-width: 200px;`, I expect that to be honored. It might be better to change the intrinsic size to account for the caret instead?

Reply:
The developer can't set a min width as this is for the inner element of the input rather than the input itself. I will make sure it works as expected though. This is how chromium have implemented this fwiw

Reply:
Ah right, targeting the inner element might be more reasonable then.